synapse/lib/socket.py

```diff
@@ -253,7 +253,7 @@ def accept(self):
         except Exception as e:
             return None, None
 
-        sock = Socket(sock, accept=True)
+        sock = self.__class__(sock, accept=True)
```

Comment:
doing this has a potential 5% performance hit

Reply:
A small hit on socket accept should be no biggie